README.md

```diff
@@ -53,3 +53,15 @@ box. We added support for additional fingerprint and bad header sources in the c
 you want to have it periodically update the fingerprint and bad header files it uses to analyze traffic.
 
 The intended entrypoint to the mitmengine package is through the `Processor.Check` function, which takes a user agent and client request fingerprint, and returns a mitm detection report. Additional API functions will be added in the future to allow for adding new signatures to a running process, for example.
+
+## How to Contribute
+
+As browser and middleware fingerprints quickly become outdated, we are actively seeking to update the fingerprint repository with new samples. To contribute fingerprint samples, please submit a PR with the following:
+- Pcap (generated by `tcpdump` or `wireshark`, for example) containing a Client Hello generated by the TLS client to fingerprint at `testdata/pcaps/<desc>/handshake.pcap`, where `<desc>` is a unique and descriptive name for the pcap sample
+- (Optional) Pcap containing an unencrypted HTTP request generated by the HTTP client to fingerprint at `testdata/pcaps/<desc>/header.pcap`
+- Line added to `testdata/fingerprint_metadata.jsonl` with the following fields (any unknown fields can be left blank or omitted):
+```
+{ "desc": "", "comment": "", "handshake_pcap": "", "header_pcap": "", "ua_fingerprint": {"raw_ua": "", "os": "", "os_version": "", "device": "", "platform": "", "browser": "", "browser_version": ""}, "middleware_fingerprint": { "name": "", "type": "" }}
```

Comment:
Can we provide them a link to uasurfer and instructions on how to fill out the fields in "ua_fingerprint"? IIRC fields like "os", "os_version", and so forth are meant to have the numeric format given by uasurfer.


For middleware fingerprint, we should also maintain a list of the possible types of middleware that the submitter can refer to.

Reply:
Oh! I see the fields in the .jsonl actually have full names! Ignore my first point--the middleware fingerprint comment still stands, though!

Reply:
Sure! I think the easiest way to add this right now is to just add the descriptions to the fields in the readme--I'll go ahead and add that now.